### PR TITLE
Improved development flow

### PR DESCRIPTION
**Pull request by jbovenschen**
A proposal / proof of concept for a new development flow.
- Uses webpack instead of browserify
- Uses UMD (So it will be compatible with most of the popular script loaders)
- While developing the content of the dist folder will not be changed
- Code linting while developing

### CODE REVIEW

File: webpack.config.base.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    output: {
+      library: 'Isomer',
+      libraryTarget: 'umd',
+      path: '/',
+      filename: (options.dev ? '[name].js' : (options.minify ? './dist/[name].min.js': './dist/[name].js')),
+      sourceMapFilename: '[file].map'
+    },
+
+    module: {
+        preLoaders: [{
```
Comment (jdan) [inline webpack.config.base.js]: Tab spacing consistency please :)

Comment (jbovenschen) [inline webpack.config.base.js]: Created a pull request to fast :-1:.

**Comment by jdan** [comment]
Really great stuff, thank you so much!

Real quick - could you explain the benefits of webpack over browserify? And what's UMD and where can I see those changes in this diff?

And one more - how does the dev-server work?

File: webpack.config.base.js (hunk 1 above)
Comment (jdan) [inline webpack.config.base.js]: No worries - not a huge deal. Thanks for fixing so quickly.

**Comment by jbovenschen** [comment]
No thanks! Just did it because i like to work with webpack and use it with a babel loader like i mentioned in #71 never used that with browserify myself.

Webpack once setup is easier to use and more plug and play (my opinion) and it has the possibility to use the ES6 module transpiler. It also does not regenerate the whole bundle only the chunks which are changed (Improvement when developing).

Here is some more info about UMD (https://github.com/umdjs/umd)
This is handled by webpack with defining the libraryTarget in the config.

The webpack dev server just serves all content in the test folder. And add the current bundles to it's assets. (Like beefy did).

File: gulpfile.js [hunk 2]
```diff
@@ -25,25 +24,35 @@ gulp.task('dist', function () {
...
-    .pipe(uglify())
+  var minified = gulp.src('./index.js')
+    .pipe(gulpWebpack(webpackConfig({minify: true})))
+    .pipe(buffer())
     .pipe(header(bannerMin, opts))
-    .pipe(rename('isomer.min.js'))
-    .pipe(gulp.dest('./dist'));
+    .pipe(gulp.dest('./'));
 
+  return es.concat(uncompressed, minified);
```
Comment (jdan) [inline gulpfile.js]: Ha, cool pattern!

**Comment by jdan** [comment]
Just checked it out, everything looks good.

Super impressive stuff, thanks again for doing this!

**Comment by jdan** [comment]
@jbovenschen I've added you as a collaborator on this repo. Feel free to close issues, merge PRs, etc. as you see fit. For any big changes I'd prefer if we continued to open PRs for each other, but I'll let you make that call :)

No pressure or anything, just figured I'd open up the gates after seeing your enthusiasm for this project. Looking forward to working with you!

**Comment by jbovenschen** [comment]
Great! Thanks!

**Comment by jbovenschen** [comment]
It is in the webpack config. You will be notified when there is an issue.

**Pull request merged**